src/platform/Zephyr/GenericThreadStackManagerImpl_Zephyr.h

```diff
@@ -77,64 +77,51 @@ class GenericThreadStackManagerImpl_Zephyr : public GenericThreadStackManagerImp
 extern template class GenericThreadStackManagerImpl_Zephyr<ThreadStackManagerImpl>;
 
 template <class ImplClass>
-CHIP_ERROR GenericThreadStackManagerImpl_Zephyr<ImplClass>::_InitThreadStack(void)
+CHIP_ERROR GenericThreadStackManagerImpl_Zephyr<ImplClass>::_InitThreadStack()
 {
     return GenericThreadStackManagerImpl_OpenThread<ImplClass>::DoInit(openthread_get_default_instance());
 }
 
 template <class ImplClass>
-CHIP_ERROR GenericThreadStackManagerImpl_Zephyr<ImplClass>::_StartThreadTask(void)
+CHIP_ERROR GenericThreadStackManagerImpl_Zephyr<ImplClass>::_StartThreadTask()
 {
     // Intentionally empty.
     return CHIP_NO_ERROR;
 }
 
 template <class ImplClass>
-void GenericThreadStackManagerImpl_Zephyr<ImplClass>::_LockThreadStack(void)
+void GenericThreadStackManagerImpl_Zephyr<ImplClass>::_LockThreadStack()
 {
-    if (k_thread_priority_get(k_current_get()) >= 0 || CONFIG_MP_NUM_CPUS != 1)
-    {
-        ChipLogError(DeviceLayer, "No locking is available. All calls to ThreadStackManager must come from a cooperative task.");
-        ChipLogError(DeviceLayer, "Offending task: %s", k_thread_name_get(k_current_get()));
-    }
+    openthread_api_mutex_lock(openthread_get_default_context());
 }
 
 template <class ImplClass>
-bool GenericThreadStackManagerImpl_Zephyr<ImplClass>::_TryLockThreadStack(void)
+bool GenericThreadStackManagerImpl_Zephyr<ImplClass>::_TryLockThreadStack()
 {
-    if (k_thread_priority_get(k_current_get()) >= 0 || CONFIG_MP_NUM_CPUS != 1)
-    {
-        ChipLogError(DeviceLayer, "No locking is available. All calls to ThreadStackManager must come from a cooperative task.");
-        ChipLogError(DeviceLayer, "Offending task: %s", k_thread_name_get(k_current_get()));
-    }
-
-    return true;
+    // Unfortunately, there's no openthread_api_mutex_try_lock() in Zephyr, so until it's contributed we must use the low-level API
```

Comment:
nit: I guess the word "Unfortunately" is not needed.

Reply:
Ok. I just expressed my feelings ;)